lib/cosmos/dart/lib/dart_reducer_worker_thread.rb

```diff
@@ -0,0 +1,252 @@
+            if job_type == :MINUTE
+              value = row_to_reduce.read_attribute(item_name)
+              min_sample = value
+              max_sample = value
+              avg_sample = value
+              if value.nil?
+                Cosmos::Logger.error("#{item_name} is nil in #{row_to_reduce.class}:#{row_to_reduce.id}")
+                next
+              end
+            else # :HOUR or :DAY
+              # We're processing hour or day data so we're reducing previously reduced data
+              # thus there are min, max, and average values to read
+              min_sample = row_to_reduce.read_attribute(min_item_name)
+              max_sample = row_to_reduce.read_attribute(max_item_name)
+              avg_sample = row_to_reduce.read_attribute(avg_item_name)
+              stddev_sample = row_to_reduce.read_attribute(stddev_item_name)
+              if min_sample.nil?
+                Cosmos::Logger.error("#{min_item_name} is nil in #{row_to_reduce.class}:#{row_to_reduce.id}")
+                next
+              end
+              if max_sample.nil?
+                Cosmos::Logger.error("#{max_item_name} is nil in #{row_to_reduce.class}:#{row_to_reduce.id}")
+                next
+              end
+              if avg_sample.nil?
+                Cosmos::Logger.error("#{avg_item_name} is nil in #{row_to_reduce.class}:#{row_to_reduce.id}")
+                next
+              end
+              if stddev_sample.nil?
+                Cosmos::Logger.error("#{stddev_item_name} is nil in #{row_to_reduce.class}:#{row_to_reduce.id}")
+                next
+              end
+            end
+
+            if nan_value?(min_sample)
+              min_nan_found = true
+            else
+              if !min_value or min_sample < min_value
+                min_value = min_sample
+              end
+            end
+
+            if nan_value?(max_sample)
+              max_nan_found = true
+            else
+              if !max_value or max_sample > max_value
+                max_value = max_sample
+              end
+            end
+
+            if nan_value?(avg_sample)
+              avg_nan_found = true
+            else
+              values << avg_sample
+              # MINUTE data is reducing the decommutated values
+              if job_type == :MINUTE
+                avg_value += avg_sample
+                total_samples += 1
+              else # :HOUR or :DAY
+                # Weight the average by multiplying by the number of samples
+                avg_value += (avg_sample * row_to_reduce.num_samples)
+                total_samples += row_to_reduce.num_samples
+              end
+            end
+          end
+          avg_value = avg_value.to_f / total_samples if total_samples != 0
+          min_value = Float::NAN if min_nan_found and !min_value
+          max_value = Float::NAN if max_nan_found and !max_value
+          mean, stddev_value = Math.stddev_population(values)
```

Comment:
@ryanatball I calculate the standard deviation on the population of the averaged values